cluster-provision/centos9/vmcli/cmd/utils/env_test.go

```diff
@@ -0,0 +1,166 @@
+			BeforeEach(func() {
+				os.Setenv(utils.NodeNumKey, "19")
+			})
+
+			It("returns the correct node number", func() {
+				nodeNb, err := envUtil.GetNodeNb()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(nodeNb).To(Equal(19))
+			})
+		})
+	})
+
+	Describe("Waiting for the TAP interface", func() {
+		When("the interface does not exist", func() {
+			const tapNum = 1
+
+			It("times out", func() {
+				err := envUtil.WaitForTap(tapNum, 1)
+				Expect(err).To(MatchError(os.ErrNotExist))
+			})
+		})
+
+		When("the interface exists", func() {
+			const tapNum = 1
+
+			BeforeEach(func() {
+				createMockDir(fs, filepath.Join(utils.NetPath, fmt.Sprintf(utils.TapName, tapNum)))
+			})
+
+			It("succeeds", func() {
+				err := envUtil.WaitForTap(tapNum, 1)
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+	})
+
+	Describe("Checking for a rootless environment", func() {
+		When("the file does not exist", func() {
+			It("returns that the environment is not rootless", func() {
+				rootless, err := envUtil.IsRootless()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(rootless).To(BeFalse())
+			})
+		})
+
+		When("the file is empty", func() {
+			BeforeEach(func() {
+				createMockFile(fs, utils.ContainerenvPath)
+			})
+
+			It("returns that the environment is not rootless", func() {
+				rootless, err := envUtil.IsRootless()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(rootless).To(BeFalse())
+			})
+		})
+
+		When("the status is not specified in the file", func() {
+			BeforeEach(func() {
+				createMockFileWithData(fs, utils.ContainerenvPath, "engine=\"podman-4.3.1\"\n")
+			})
+
+			It("returns that the environment is not rootless", func() {
+				rootless, err := envUtil.IsRootless()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(rootless).To(BeFalse())
+			})
+		})
+
+		When("the rootless field has an invalid value", func() {
+			BeforeEach(func() {
+				createMockFileWithData(fs, utils.ContainerenvPath, "engine=\"podman-4.3.1\"\nrootless=invalid\n")
+			})
+
+			It("returns that the environment is not rootless", func() {
+				rootless, err := envUtil.IsRootless()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(rootless).To(BeFalse())
+			})
+		})
+
+		When("the rootless field has an invalid number", func() {
+			BeforeEach(func() {
+				createMockFileWithData(fs, utils.ContainerenvPath, "engine=\"podman-4.3.1\"\nrootless=2\n")
+			})
+
+			It("returns that the environment is not rootless", func() {
+				rootless, err := envUtil.IsRootless()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(rootless).To(BeFalse())
+			})
+		})
+
+		When("the environment is not rootless", func() {
+			BeforeEach(func() {
+				createMockFileWithData(fs, utils.ContainerenvPath, "engine=\"podman-4.3.1\"\nrootless=0\n")
+			})
+
+			It("returns that the environment is not rootless", func() {
+				rootless, err := envUtil.IsRootless()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(rootless).To(BeFalse())
+			})
+		})
```

Comment:
This looks like it could be reduce and merged together into a DescribeTable. You can pass a function as parameter and execute it based on the different entries